src/main/java/com/agata/jeeshop/services/CartService.java

```diff
@@ -66,22 +63,27 @@ public  CartDto addToCart(AddProductToCartRequest addProductToCartRequest){
         return cartMapper.toDto(cart);
     }
 
-    public String calculateTotalAmount(Long id){
+    public String calculateTotalAmount(Long id) {
 
         CartDto cartDto = this.findById(id);
-        Double totalAmount=0.;
-        for (CartItemDto cartItem:cartDto.getItems()
-             ) {
-            totalAmount+=cartItem.getProductPrice();
+        Double totalAmount = 0.;
+        for (CartItemDto cartItem : cartDto.getItems()
+        ) {
+            totalAmount += cartItem.getProductPrice();
         }
         this.findById(id).setTotalAmount(totalAmount);
 
 
         return String.format("%.2f", totalAmount);
     }
 
-   public CartItemDto removeProduct(CartItemDto cartItemDto) {
-        Cart cart = cartRepository.save(cartMapper.fromDto(cartItemDto));
-        return cartMapper.toDto(cart);
+    public void removeProduct(Long cartId, Long id) {
+        Cart cart = cartRepository.getOne(cartId);
+        for (CartItem item : cart.getItems()) {
+            if (item.getId() == id)
+                cartItemRepository.delete(item);
+            else
+                throw new RuntimeException("Cart item with id: " + cart.getItems() + " not found");
```

Comment:
Wyjątkiem rzuć dopiero po pętli, tutaj robisz tak, że jeżęli są np. 2 przedmioty w koszyku i pierwszy nie jest z tym id, które sprawdzamy to zawsze zostanie rzucony wyjątek. Może póki co darujmy sobie walidację i wywal w ogole to rzucanie wyjątku